components/TextFields/src/private/MDCTextInputCommonFundament.m

```diff
@@ -344,26 +348,37 @@ - (void)setupUnderlineLabels {
   [NSLayoutConstraint
       activateConstraints:@[ labelSpacing, _leadingUnderlineLeading, _trailingUnderlineTrailing ]];
 
-  NSLayoutConstraint *leadingBottom = [NSLayoutConstraint constraintWithItem:_leadingUnderlineLabel
-                                                                   attribute:NSLayoutAttributeBottom
-                                                                   relatedBy:NSLayoutRelationEqual
-                                                                      toItem:_textInput
-                                                                   attribute:NSLayoutAttributeBottom
-                                                                  multiplier:1
-                                                                    constant:0];
-  leadingBottom.priority = UILayoutPriorityDefaultLow;
-
-  NSLayoutConstraint *trailingBottom =
-      [NSLayoutConstraint constraintWithItem:_trailingUnderlineLabel
+  NSLayoutConstraint *leadingTop =
+  [NSLayoutConstraint constraintWithItem:_underline
+                               attribute:NSLayoutAttributeBottom
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:_leadingUnderlineLabel
+                               attribute:NSLayoutAttributeTop
+                              multiplier:1
+                                constant:-MDCTextInputHalfPadding];
+  leadingTop.priority = UILayoutPriorityDefaultLow;
+
+  NSLayoutConstraint *leadingBottom =
+      [NSLayoutConstraint constraintWithItem:_leadingUnderlineLabel
                                    attribute:NSLayoutAttributeBottom
-                                   relatedBy:NSLayoutRelationEqual
+                                   relatedBy:NSLayoutRelationGreaterThanOrEqual
                                       toItem:_textInput
                                    attribute:NSLayoutAttributeBottom
                                   multiplier:1
-                                    constant:0];
-  trailingBottom.priority = UILayoutPriorityDefaultLow;
+                                    constant:-MDCTextInputHalfPadding];
+  leadingBottom.priority = UILayoutPriorityDefaultLow;
 
-  [NSLayoutConstraint activateConstraints:@[ leadingBottom, trailingBottom ]];
+  NSLayoutConstraint *trailingTop =
+  [NSLayoutConstraint constraintWithItem:_underline
+                               attribute:NSLayoutAttributeBottom
+                               relatedBy:NSLayoutRelationEqual
+                                  toItem:_trailingUnderlineLabel
+                               attribute:NSLayoutAttributeTop
+                              multiplier:1
+                                constant:-MDCTextInputHalfPadding];
```

Comment:
I prefer a `-1 * [whatever]` for negating constants. It's easy to miss the hyphen by itself.